src/simplify_algebra.cpp

```diff
@@ -708,6 +710,109 @@ struct find_rsqrt
     }
 };
 
+auto slice_reshape_transpose_ops(const std::string& input, const std::string& reshape)
+{
+    return match::name("transpose")(
+        match::arg(0)(match::name("reshape")(match::arg(0)(match::name("contiguous")(
+                                                 match::arg(0)(match::name("slice").bind(input)))))
+                          .bind(reshape)));
+}
+
+struct find_slice_reshape_transpose
```

Comment:
This should be split into two separate classes. There should be a `find_slice_reshape` that fuses the reshape across slices, and then a `find_slice_transpose` that fuses the transpose across slices.

Reply:
fixed accordingly